tmt/steps/prepare/install.py

```diff
@@ -92,164 +360,36 @@ def wake(self, keys=None):
-                    self.debug(f"Found rpm '{filename}'.", level=3)
-                    local_packages.append(os.path.join(directory, filename))
-
-        # Prepare the install options
-        options = '-y'
-        for package in self.get('exclude'):
-            options += " --exclude " + tmt.utils.quote(package)
-
-        # Install from local filesystem
-        if local_packages:
-            rpms_directory = os.path.join(self.step.workdir, 'rpms')
-            os.makedirs(rpms_directory)
-
-            # Copy local packages into workdir, push to guests
-            for package in local_packages:
-                self.verbose(os.path.basename(package), shift=1)
-                self.debug(f"Copy '{package}' to '{rpms_directory}'.", level=3)
-                shutil.copy(package, rpms_directory)
-            for guest in self.step.plan.provision.guests():
-                guest.push()
-
-            # Use both dnf install/reinstall to get all packages refreshed
-            # FIXME Simplify this once BZ#1831022 is fixed/implemeted.
-            guest.execute(f"{command} install {options} {rpms_directory}/*")
-            guest.execute(f"{command} reinstall {options} {rpms_directory}/*")
-            summary = fmf.utils.listed(local_packages, 'local package')
-            self.info('total', f"{summary} installed", 'green')
-
-        # Install from repositories
-        if repo_packages:
-            packages = self.prepare_packages(repo_packages, title="package")
-            # Extra ignore/check for yum to workaround BZ#1920176
-            check = f'rpm -q --whatprovides {packages}'
-            if 'yum' in command:
-                yum_check = " || true" if skip else f" && {check}"
-            else:
-                yum_check = ""
-            # Check and install
-            guest.execute(
-                f"{check} || {command} install {options} "
-                f"{packages}{yum_check}")
-
-        # Install debug{info,source} from repos
-        if debuginfo_packages:
-            packages = self.prepare_packages(
-                debuginfo_packages, title="debuginfo")
-            # Make sure debuginfo-install is present on the target system
-            guest.execute(f"{command} install -y /usr/bin/debuginfo-install")
-            guest.execute(f"debuginfo-install -y {packages}")
+            try:
+                guest.execute('rpm -q dnf')
+                installer = InstallDnf(
+                    parent=self,
+                    guest=guest,
+                    package=self.get("package"),
+                    pkg_mgr='dnf',
+                    pkg_data=self.data)
+            except tmt.utils.RunError:
+                installer = InstallDnf(
+                    parent=self,
+                    guest=guest,
+                    package=self.get("package"),
+                    pkg_mgr='yum',
+                    pkg_data=self.data)
+                plugin = 'yum-plugin-copr'
```

Comment:
1. `plugin` seems to slip out of `InstallDnf` call
2. `plugin` shouldn't be needed at this place, as I mentioned few comments before, the different value of `pkg_mgr` and `plugin` needed to "switch" `InstallDnf` to work with `yum` calls for `InstallYum` which would call superclass's `__init__()` with necessary values.

```python
class InstallBase(tmt.utils.Common):
    def __init__(self, parent, guest, package, pkg_data, pkg_mgr, plugin=None):
        ...

class InstallDnf(InstallBase):
    def __init__(self, parent, guest, package, pkg_data, pkg_mgr='dnf', plugin='dnf-plugins-core'):
        super().__init__(parent, guest, package, pkg_data, pkg_mgr, plugin=plugin)
        ...

class InstallYum(InstallDnf):
    def __init__(self, parent, guest, package, pkg_data):
        super().__init__(parent, guest, package, pkg_data, pkg_mgr='yum', plugin='yum-plugin-copr')
        ...

installer = InstallDnf(parent=self, guest=guest, package=self.get("package"), pkg_data=self.data)
installer = InstallYum(parent=self, guest=guest, package=self.get("package"), pkg_data=self.data)
```

Really, what the right plugin should be is really none of my business, I fully expect `InstallDnf` to know that since it's **the** installer class responsible for performing installations with `dnf`.